diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,132 @@
+package(default_visibility = ["//visibility:public"])
+
+proto_library(
+    name = "type_proto",
+    srcs = ["proto/substrait/type.proto"],
+    strip_import_prefix = "proto",
+    deps = [
+        "@protobuf//:empty_proto",
+    ],
+)
+
+cc_proto_library(
+    name = "type_cc_proto",
+    deps = [":type_proto"],
+)
+
+proto_library(
+    name = "parameterized_types_proto",
+    srcs = ["proto/substrait/parameterized_types.proto"],
+    strip_import_prefix = "proto",
+    deps = [
+        ":type_proto",
+    ],
+)
+
+cc_proto_library(
+    name = "parameterized_types_cc_proto",
+    deps = [":parameterized_types_proto"],
+)
+
+proto_library(
+    name = "type_expressions_proto",
+    srcs = ["proto/substrait/type_expressions.proto"],
+    strip_import_prefix = "proto",
+    deps = [
+        ":type_proto",
+    ],
+)
+
+cc_proto_library(
+    name = "type_expressions_cc_proto",
+    deps = [":type_expressions_proto"],
+)
+
+proto_library(
+    name = "extensions_proto",
+    srcs = ["proto/substrait/extensions/extensions.proto"],
+    strip_import_prefix = "proto",
+    deps = [
+        "@protobuf//:any_proto",
+    ],
+)
+
+cc_proto_library(
+    name = "extensions_cc_proto",
+    deps = [":extensions_proto"],
+)
+
+proto_library(
+    name = "algebra_proto",
+    srcs = ["proto/substrait/algebra.proto"],
+    strip_import_prefix = "proto",
+    deps = [
+        ":extensions_proto",
+        ":type_proto",
+        "@protobuf//:any_proto",
+    ],
+)
+
+cc_proto_library(
+    name = "algebra_cc_proto",
+    deps = [":algebra_proto"],
+)
+
+proto_library(
+    name = "plan_proto",
+    srcs = ["proto/substrait/plan.proto"],
+    strip_import_prefix = "proto",
+    deps = [
+        ":algebra_proto",
+        ":extensions_proto",
+    ],
+)
+
+cc_proto_library(
+    name = "plan_cc_proto",
+    deps = [":plan_proto"],
+)
+
+proto_library(
+    name = "capabilities_proto",
+    srcs = ["proto/substrait/capabilities.proto"],
+    strip_import_prefix = "proto",
+)
+
+cc_proto_library(
+    name = "capabilities_cc_proto",
+    deps = [":capabilities_proto"],
+)
+
+proto_library(
+    name = "extended_expression_proto",
+    srcs = ["proto/substrait/extended_expression.proto"],
+    strip_import_prefix = "proto",
+    deps = [
+        ":algebra_proto",
+        ":extensions_proto",
+        ":plan_proto",
+        ":type_proto",
+    ],
+)
+
+cc_proto_library(
+    name = "extended_expression_cc_proto",
+    deps = [":extended_expression_proto"],
+)
+
+proto_library(
+    name = "function_proto",
+    srcs = ["proto/substrait/function.proto"],
+    strip_import_prefix = "proto",
+    deps = [
+        ":parameterized_types_proto",
+        ":type_expressions_proto",
+        ":type_proto",
+    ],
+)
+
+cc_proto_library(
+    name = "function_cc_proto",
+    deps = [":function_proto"],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1,6 @@
+module(
+    name = "substrait",
+    version = "0.44.0",
+)
+
+bazel_dep(name = "protobuf", version = "32.0")
